src/Template/Element/index/bulk_actions/form_end.ctp

```diff
@@ -0,0 +1,30 @@
+<?php
+if (empty($bulkActions)) {
+    return;
+}
+
+$this->Form->templates([
+    'submitContainer' => '{{content}}',
+]);
+
+$submitButton = $this->Form->submit(__d('crud', 'Apply'), [
+    'class' => 'btn btn-success btn-bulk-apply',
+    'div' => false,
+    'name' => '_bulk',
+]);
+$this->Form->templates([
+    'inputContainer' => '<div class="form-group bulk-action-submit {{required}}">{{content}}{{help}}</div>',
+    'select' => '<div class="col-sm-10"><select name="{{name}}"{{attrs}}>{{content}}</select>' . $submitButton . '</div>',
```

Comment:
> I might be commenting on how the form helper works rather than this though - isn't it possible to switch template sets at run time, without re-creating the form helper object?

It is. FormHelper::push() and FormHelper::popup() allow you to backup current templates and restore them back. Though generally it's not required to do that explicitly. Templates passed to FormHelper::create() are reverted backup when calling FormHelper::end(). Similarly templates passsed to FormHelper::input() calls are also reverted back after the call.